proto/gnmi_ext/gnmi_ext.proto

```diff
@@ -89,3 +92,47 @@ message TimeRange {
   int64 start = 1; // Nanoseconds since the epoch
   int64 end = 2;   // Nanoseconds since the epoch
 }
+
+// The CommitID is returned by the gNMI server to indicate the ID of a commit.
+// It is returned in response to the Set Request with the CommitConfirm message
+// that had COMMIT_CONFIRMED_ACTION_START set.
+// The CommitID is used in the subsequent SetRequest/CommitConfirm messages
+// to identify commit to be confirmed, reset or rejected.
+message CommitID {
+  string id = 1;
```

Comment:
What is the level of security on the commit id here?
- Can I accidentally or maliciously confirm or reject someone else's commit?
What is the scope
- Is the commit id expected to be valid only within some scope?
What checks can a server make to see that the CONFIRM/REJECT request comes from the same client as the START (other than commit id match); can we assume the following request will come from a connection/session with the same parameters?
Can any number of commit confirms be open at once? (What is a server wanted to set a max limit)?
Should there be a max timeout, or could a server impose one?  (If it can, should the  CommitID message also include a duration in the response so the client knows what time they have?  Similarly would it be useful to return the server default duration for the case where 0 is input?)

Reply:
1. a user executing gNMI RPC is authenticated using the [standard auth mechanisms](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-authentication.md). It might depend on the implementation, but one can expect that a commit confirmed started by user U1 can only be confirmed by U1 user.
2. I am not aware of NOSes that support multiple parallel commits. If there are NOSes that support multiple parallel commits this spec must be adjusted.

Reply:
The proposed commit-confirmed workflow assumes that once the commit-confirmed started, the lock is put in place preventing any other commits from happening. 

Any subsequent SetRequest without commit-confirmed extension (a regular SetRequest) will yield an error. This is not explained in https://github.com/openconfig/reference/pull/198 and you made a great point that it needs to be clarified.

Because no parallel commit-confirmed are allowed, it is implicitly assumed that the running datastore is locked for any other commits until the commit in question is confirmed or rejected

Reply:
Great - yes please clarify in the document, but also I think it would be better to make this explict here with the action names or at least the comments next to the action names:
START has to also LOCK
REJECT/CONFIRM have to UNLOCK
TIMEOUT has to be handled as REJECT (rollback config and also UNLOCK)

Reply:
I will defer for now and wait for OC/Google comments on this PR since it can be they would like to proceed with their version which is in https://github.com/openconfig/gnmi/pull/155